lib/Doctrine/ORM/Tools/SchemaTool.php

```diff
@@ -910,4 +911,42 @@ public function getUpdateSchemaSql(array $classes, $saveMode = false)
 
         return $schemaDiff->toSql($this->platform);
     }
+
+    /**
+     * Creates the schema from the database, ensuring tables from the target schema are whitelisted for comparison.
+     */
+    private function createSchemaForComparison(Schema $toSchema) : Schema
+    {
+        // check and memorize if we are using dbal >= 3.0
+        static $hasFilterExpressionMethod;
+        $hasFilterExpressionMethod = $hasFilterExpressionMethod ?? method_exists(Configuration::class, 'getFilterSchemaAssetsExpression');
```

Comment:
Is this really needed? This function should be only called a few times, this looks like unnecessary optimization to me. Also we don't use `static $foo` in our code to declare global variables.

Reply:
static variable removed